### ISSUE TITLE
OSClass do match all CPE's but not the current one

### ISSUE THREAD

**Issue by evait-security** [sophsec/ruby-nmap#43]
While i list the CPE´s from the OSClass i found that the OSClass do not list the current cpe´s from the current node but list all cpes from the global node of the xml. 

my Code to list the current cpe:
```ruby
puts "OS: [#{host.os.to_a}"
host.os.each do |osMatch|
      puts "osMatch: #{osMatch}"
      puts "name: #{osMatch.name}"
      puts "accuracy: #{osMatch.accuracy}"
end
host.os.classes.to_a.each do |osClass|
      puts 'cpe class:'
      puts "#{osClass} V. #{osClass.gen}"
      osClass.cpe.to_a.each do |curCpe|
            puts "cpe: #{curCpe}"
      end
end
```
the output:
```
OS: [[#<struct Nmap::OSMatch name="Linux 2.6.32", accuracy=100>]
osMatch: Linux 2.6.32 (100%)
name: Linux 2.6.32
accuracy: 100
cpe class:
general purpose Linux (100%) V. 2.6.X
cpe: cpe:/o:linux:linux_kernel:3
cpe: cpe:/o:linux:linux_kernel:4
cpe: cpe:/o:microsoft:windows_server_2012
cpe: cpe:/o:linux:linux_kernel:2.6.32

```
the part of the xml:
```
<os><portused state="open" proto="tcp" portid="53"/>
<portused state="closed" proto="tcp" portid="21"/>
<portused state="closed" proto="udp" portid="36275"/>
<osmatch name="Linux 2.6.32" accuracy="100" line="49805">
<osclass type="general purpose" vendor="Linux" osfamily="Linux" osgen="2.6.X" accuracy="100"><cpe>cpe:/o:linux:linux_kernel:2.6.32</cpe></osclass>
</osmatch>
</os>

```

**Comment by postmodern** [comment]
I think this was fixed in 1404475c7fb879a71f661a3acdad77b453152a0e, aka #37? It appears I need to release a new version.

**Comment by postmodern** [comment]
Fixed in 0.9.0.
